homeassistant/data_entry_flow.py

```diff
@@ -717,17 +718,17 @@ def async_start_progress(
         """Start in progress task if not started."""
 
         async def _send_event_when_done(fut: asyncio.Future[None]) -> None:
-            if not fut.done():
-                try:
-                    await fut
-                except Exception as err:  # pylint: disable=broad-exception-caught
-                    fut.set_exception(err)
+            with suppress(BaseException):
```

Comment:
If we dont suppress exceptions here but use try... finally, we only need one task.

Reply:
This is not correct, any exception raised is relevant only for the task passed to us be a config flow, we don't care about it and no one will retrieve it from this task